### PR TITLE
print correct number of ballots (closes #2504)

### LINKED ISSUE #2504: Use number of ballot config for motions/assignments
Use config 'motions_pdf_ballot_papers_selection' for generating motion ballot PDF, see
https://github.com/OpenSlides/OpenSlides/blob/master/openslides/motions/config_variables.py#L190

Same for new assignment ballot PDF.

### PR DESCRIPTION
**Pull request by matakuka**
The pdf printing now respects the `..._pdf_ballot_papers_number` number; tested with up to 10000 ballots. 

It is not ready yet, as the config also gives the options:
- NUMBER_OF_DELEGATES
- NUMBER_OF_ALL_PARTICIPANTS

I am unsure how to get these numbers. Also, the printing of the last page is missing one horizontal line.

### CODE REVIEW

File: openslides/assignments/static/js/assignments/pdf.js [hunk 1]
```diff
@@ -367,58 +368,113 @@ angular.module('OpenSlidesApp.assignments.pdf', ['OpenSlidesApp.core.pdf'])
...
 
             var createContentTable = function() {
-
-                var tableBody = [];
+                // first, determine how many ballots we need
+                var amount;
+                var amount_method = Config.get('assignments_pdf_ballot_papers_selection').value;
+                switch (amount_method) {
+                    case 'NUMBER_OF_ALL_PARTICIPANTS':
+                        //TODO: check if getAll really includes all users
```
Comment (emanuelschuetze) [inline openslides/assignments/static/js/assignments/pdf.js]: yes, I checked it. You can remove this todo.

File: openslides/assignments/static/js/assignments/pdf.js [hunk 2]
```diff
@@ -367,58 +368,113 @@ angular.module('OpenSlidesApp.assignments.pdf', ['OpenSlidesApp.core.pdf'])
...
+                        amount = User.getAll().length;
+                        break;
+                    case 'NUMBER_OF_DELEGATES':
+                        //TODO: See above, same problem, but now we need all that are in a group
```
Comment (emanuelschuetze) [inline openslides/assignments/static/js/assignments/pdf.js]: You can filter the users with js-data (not tested):
`User.filter({where: { groups_id: 2}}).length`

see e.g. in  [agenda/projector.js](https://github.com/OpenSlides/OpenSlides/blob/ac60fb53daf8cc51a49cd3d834ef9f038779987f/openslides/agenda/static/js/agenda/projector.js#L107)

File: openslides/motions/static/js/motions/pdf.js [hunk 3]
```diff
@@ -367,21 +369,75 @@ angular.module('OpenSlidesApp.motions.pdf', ['OpenSlidesApp.core.pdf'])
...
-                    ],
-                },
-                layout: PDFLayout.getBallotLayoutLines()
-            }];
+            var content = [];
+            var amount;
+            var amount_method = Config.get('motions_pdf_ballot_papers_selection').value;
+            switch (amount_method) {
+                    case 'NUMBER_OF_ALL_PARTICIPANTS':
+                        //TODO: check if getAll really includes all users
```
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/pdf.js]: see above

File: openslides/motions/static/js/motions/pdf.js [hunk 4]
```diff
@@ -367,21 +369,75 @@ angular.module('OpenSlidesApp.motions.pdf', ['OpenSlidesApp.core.pdf'])
...
+                        amount = User.getAll().length;
+                        break;
+                    case 'NUMBER_OF_DELEGATES':
+                        //TODO: See above, same problem, but now we need all that are in a group
```
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/pdf.js]: see above

**Comment by matakuka** [comment]
@emanuelschuetze done now as suggested.

**Comment by normanjaeckel** [comment]
Please add CHANGELOG entry and rebase.

**Comment by emanuelschuetze** [comment]
@matakuka ping

**Comment by matakuka** [comment]
@emanuelschuetze  done

**Comment by emanuelschuetze** [comment]
Thanks

**Pull request merged**